app/modules/media/media_controller.py

```
@@ -0,0 +1,271 @@
+                file=file,
+                file_name=file.filename,
+                mime_type=file.content_type,
+                message_id=message_id,
+            )
+
+            logger.info(f"User {self.user_id} uploaded image: {result.id}")
+            return result
+
+        except MediaServiceError as e:
+            logger.error(f"Media service error: {str(e)}")
+            raise HTTPException(status_code=500, detail=str(e))
```

Ruff (0.11.9)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)